stock_inventory_discrepancy/tests/test_inventory_discrepancy.py

```diff
@@ -0,0 +1,199 @@
+        )
+        self.test_loc = self.obj_location.create(
+            {"name": "Test Location", "usage": "internal", "discrepancy_threshold": 10}
+        )
+        self.test_wh = self.obj_warehouse.create(
+            {"name": "Test WH", "code": "T", "discrepancy_threshold": 20}
+        )
+        self.obj_location._parent_store_compute()
+
+        # Create Stock manager able to force validation on inventories.
+        group_stock_man = self.env.ref("stock.group_stock_manager")
+        group_inventory_all = self.env.ref(
+            "stock_inventory_discrepancy.group_stock_inventory_validation_always"
+        )
+        group_employee = self.env.ref("base.group_user")
+
+        self.manager = self.env["res.users"].create(
+            {
+                "name": "Test Manager",
+                "login": "manager",
+                "email": "test.manager@example.com",
+                "groups_id": [(6, 0, [group_stock_man.id, group_inventory_all.id])],
+            }
+        )
+        group_stock_user = self.env.ref("stock.group_stock_user")
+        group_inventory = self.env.ref(
+            "stock_inventory_discrepancy.group_stock_inventory_validation"
+        )
+        self.user = self.env["res.users"].create(
+            {
+                "name": "Test User",
+                "login": "user",
+                "email": "test.user@example.com",
+                "groups_id": [(6, 0, [group_stock_user.id, group_inventory.id])],
+            }
+        )
+
+        self.user_2 = self.env["res.users"].create(
+            {
+                "name": "Test User 2",
+                "login": "user_2",
+                "email": "test2.user@example.com",
+                "groups_id": [(6, 0, [group_stock_user.id, group_inventory_all.id])],
+            }
+        )
+
+        self.no_user = self.env["res.users"].create(
+            {
+                "name": "No User",
+                "login": "no_user",
+                "email": "test.no_user@example.com",
+                "groups_id": [(6, 0, [group_employee.id])],
+            }
+        )
+
+        self.quant_line1 = self.obj_quant.with_context(inventory_mode=True).create(
+            {
+                "product_id": self.product1.id,
+                "quantity": 2.0,
+                "location_id": self.test_loc.id,
+            }
+        )
+        self.quant_line2 = self.obj_quant.with_context(inventory_mode=True).create(
+            {
+                "product_id": self.product2.id,
+                "quantity": 4.0,
+                "location_id": self.test_loc.id,
+            }
+        )
+
+    def test_discrepancy_validation(self):
+        """Tests the new workflow"""
+        # quant_line1 is over discrepancy but quant_line2 is not
+        self.quant_line1.write(
+            {
+                "inventory_quantity": 3.0,
+                "inventory_quantity_set": True,
+            }
+        )
+        self.quant_line1._compute_discrepancy_threshold()
```

Comment:
Calling this compute function after a quant is created will force a call to the write() method, which may cause tests to fail due to the changed behavior of the write() method in this commit: https://github.com/odoo/odoo/commit/e21c17fae8a29c8ee04883b27d761b8ce13cf5b7.

Reply:
Indeed, in test you should set inventory_mode=False in context